src/Orleans/Core/IInternalClusterClient.cs

```diff
@@ -11,5 +13,26 @@ internal interface IInternalClusterClient : IClusterClient, IInternalGrainFactor
         /// Gets the client's <see cref="IStreamProviderRuntime"/>.
         /// </summary>
         IStreamProviderRuntime StreamProviderRuntime { get; }
+
+        /// <summary>
+        /// Global pre-call interceptor function
+        /// Synchronous callback made just before a message is about to be constructed and sent by a client to a grain.
+        /// This call will be made from the same thread that constructs the message to be sent, so any thread-local settings
+        /// such as <c>Orleans.RequestContext</c> will be picked up.
+        /// The action receives an <see cref="InvokeMethodRequest"/> with details of the method to be invoked, including InterfaceId and MethodId,
+        /// and a <see cref="IGrain"/> which is the GrainReference this request is being sent through
+        /// </summary>
+        /// <remarks>This callback method should return promptly and do a minimum of work, to avoid blocking calling thread or impacting throughput.</remarks>
+        ClientInvokeCallback ClientInvokeCallback { get; set; }
+
+        /// <summary>
+        /// Event fired when connection to the cluster is lost.
+        /// </summary>
+        event ConnectionToClusterLostHandler ClusterConnectionLost;
```

Comment:
They're only there for tests and GrainClient compat. I can remove `ResponseTimeout` outright fairly easily using DI to access the `OutsideRuntimeClient` in the relevant test and in `GrainClient`.

I could use the same trick for the other two. Do you feel that's best?